src/main/java/nwoolcan/model/brewery/production/batch/review/BatchEvaluationBuilder.java

```diff
@@ -0,0 +1,154 @@
+        this.evaluations.add(new EvaluationImpl(type, score, Optional.of(notes)));
+        return this;
+    }
+
+    /**
+     * Add evaluation for a category without notes.
+     *
+     * @param type  category.
+     * @param score score.
+     * @return this.
+     */
+    public BatchEvaluationBuilder addEvaluation(final EvaluationType type, final int score) {
+        this.evaluations.add(new EvaluationImpl(type, score, Optional.empty()));
+        return this;
+    }
+
+    /**
+     * Add notes for the review.
+     *
+     * @param notes notes.
+     * @return this.
+     */
+    public BatchEvaluationBuilder addNotes(final String notes) {
+        this.notes = notes;
+        return this;
+    }
+
+    /**
+     * Add reviewer for the review.
+     *
+     * @param reviewer reviewer.
+     * @return this.
+     */
+    public BatchEvaluationBuilder addReviewer(final String reviewer) {
+        this.reviewer = reviewer;
+        return this;
+    }
+
+    /**
+     * Return whether this builder is already built.
+     * @return whether this builder is already built.
+     */
+    public boolean isBuilt() {
+        return this.built;
+    }
+
+    /**
+     * Return a {@link Result} holding a {@link BatchEvaluation} if everything went well, otherwise a
+     * {@link Result} holding an {@link Exception}.
+     *
+     * @return a {@link Result< BatchEvaluation >}
+     */
+    public Result<BatchEvaluation> build() {
+        return Result.of(this)
+            .require(BatchEvaluationBuilder::checkEvaluationsTypeValidity,
+                new IllegalArgumentException(INVALID_CATEGORIES_MESSAGE))
+            .require(BatchEvaluationBuilder::checkScoreValidity,
+                new IllegalAccessException(INVALID_SCORE_MESSAGE))
+            .require(builder -> !builder.built,
+                new IllegalStateException(BUILDER_USED_MESSAGE))
+            .peek(builder -> builder.built = true)
+            .map(builder -> new BatchEvaluationImpl(builder.batchEvaluationType,
+                builder.evaluations,
+                Optional.ofNullable(builder.reviewer),
+                Optional.ofNullable(builder.notes)));
+    }
+
+    /**
+     * Resets the builder.
+     *
+     * @param newType the {@link BatchEvaluationType} of the new review.
+     * @return this
+     */
+    public BatchEvaluationBuilder reset(final BatchEvaluationType newType) {
+        this.evaluations.clear();
+        this.notes = null;
+        this.reviewer = null;
+        this.built = false;
+        this.batchEvaluationType = newType;
+        return this;
+    }
+
+    private boolean checkEvaluationsTypeValidity() {
+        Collection<EvaluationType> catTypes = this.evaluations
+            .stream()
+            .map(Evaluation::getEvaluationType)
+            .collect(Collectors.toList());
+        return catTypes.containsAll(this.batchEvaluationType.getCategories())
```

Comment:
can't this be only one line?

Reply:
I make use of `catTypes` two times in the condition, rewriting the stream does not seem a good idea to me